pkg/controller/plan/validation.go

```diff
@@ -421,7 +446,7 @@ func (r *Reconciler) validateVM(plan *api.Plan) error {
 		if err != nil {
 			return err
 		}
-		if plan.Referenced.Secret == nil {
+		if plan.Referenced.Secret == nil && plan.Referenced.Provider.Source.Type() != api.OpenShift {
```

Comment:
a comment regarding this?
we won't need the `plan.Referenced.Secret` anywhere if the source is OCP?

Reply:
> we won't need the `plan.Referenced.Secret` anywhere if the source is OCP?

yeah, that's a good point - we should check if the source is the 'host' cluster as the latest code does